viadot/sources/genesys.py

```diff
@@ -0,0 +1,526 @@
+
+        self.logger.info("Al reports were successfully dowonload.")
+
+        if store_file_names is True:
+            self.logger.info("Successfully genetared file names list.")
+            return file_name_list
+
+    def get_analitics_url_report(self):
+        """Fetching analytics report url from json response.
+
+        Returns:
+            string: url for analytics report
+        """
+        response = handle_api_response(
+            url=f"https://api.{self.environment}/api/v2/analytics/reporting/schedules/{self.schedule_id}",
+            headers=self.authorization_token,
+        )
+        try:
+            response_json = response.json()
+            report_url = response_json.get("lastRun", None).get("reportUrl", None)
+            self.logger.info("Successfully downloaded report from genesys api")
+            return report_url
+        except AttributeError as e:
+            self.logger.error(
+                "Output data error: " + str(type(e).__name__) + ": " + str(e)
+            )
+
+    def get_all_schedules_job(self):
+        """Fetching analytics report url from json response.
+
+        Returns:
+            string: json body with all schedules jobs.
+        """
+        response = handle_api_response(
+            url=f"https://api.{self.environment}/api/v2/analytics/reporting/schedules",
+            headers=self.authorization_token,
+        )
+        try:
+            response_json = response.json()
+            self.logger.info("Successfully downloaded schedules jobs.")
+            return response_json
+        except AttributeError as e:
+            self.logger.error(
+                "Output data error: " + str(type(e).__name__) + ": " + str(e)
+            )
+
+    def schedule_report(self, data_to_post: Dict[str, Any]):
+        """POST method for report scheduling.
+
+        Args:
+            data_to_post (Dict[str, Any]): json format of POST body.
+
+        Returns:
+            bool: schedule genesys report
+        """
+        payload = json.dumps(data_to_post)
+        new_report = handle_api_response(
+            url=f"https://api.{self.environment}/api/v2/analytics/reporting/schedules",
+            headers=self.authorization_token,
+            method="POST",
+            body=payload,
+        )
+        if new_report.status_code == 200:
+            self.logger.info("Succesfully scheduled new report.")
+        else:
+            self.logger.error(f"Failed to scheduled new report. - {new_report.content}")
+            raise APIError("Failed to scheduled new report.")
+        return new_report.status_code
+
+    def generate_reporting_export(
+        self, data_to_post: Dict[str, Any], verbose: bool = False
+    ):
+        """POST method for reporting export.
+
+        Args:
+            data_to_post (Dict[str, Any]): json format of POST body.
+
+        Returns:
+            bool: schedule genesys report
```

Comment:
`new_report.status_code` is not bool value I guess. In docstring you have `Returns: bool: ...` but status_code is an integer.